### PR TITLE
Forms: Add field preview component for response view

### PR DESCRIPTION
**Pull request by CGastrell**
Part of FORMS-520
Part of FORMS-521
Part of FORMS-522
Part of FORMS-523
Part of FORMS-524
Part of FORMS-525
Part of FORMS-526
Part of FORMS-527
Part of FORMS-528
Part of FORMS-529
Part of FORMS-530

## Proposed changes:
* Add new `FieldPreview` component for rendering form response fields in the dashboard
* Introduce new `collection` format for API field responses with structured field properties
* Add type definitions for legacy and new field formats (`ResponseField`, `FieldType`)
* Render appropriate icons for each field type using block definitions
* Properly handle different value types (arrays, files, image-select, emails, URLs, phone numbers)
* Scope styles to prevent old format rendering from affecting new FieldPreview styles

| Before | After |
|--------|--------|
| image | image |

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
8TmDfrB54NU4Rzj19Qs9Tg-fi-4935_76204
p1769009515362959-slack-C052XEUUBL4

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Go to the Jetpack Forms dashboard 
* Submit a form with various field types (text, email, phone, URL, file upload, image select, checkbox, radio, etc.)
* View the form responses in the dashboard
* Verify that each field displays with the correct icon matching its field type
* Verify that field values render correctly:
  - Email fields show clickable mailto links
  - Phone fields show clickable tel links
  - URL fields show external links
  - File uploads show file previews
  - Image selects show image choices
  - Arrays (multiple choice) display as comma-separated values
* Confirm styles are properly scoped and don't affect other response view elements

### CODE REVIEW

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+.jp-forms__field-preview-icon {
+	color: var(--jp-gray-50, #646970);
+	flex-shrink: 0;
+
+	svg {
+		display: block;
+		width: 24px;
+		height: 24px;
+		fill: currentColor;
```
Comment (simison) [inline projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss]: Better throw `Icon` component there and you don't need the styles here + some of the Element handling you had might get simplified too, since that component "eats" many things.

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+			return <FieldEmail email={ stringValue } />;
+		}
+
+		// Phone numbers
+		if ( fieldType === 'phone' || fieldType === 'telephone' ) {
+			return <a href={ `tel:${ stringValue }` }>{ stringValue }</a>;
+		}
+
+		if ( fieldType === 'url' ) {
+			return <ExternalLink href={ stringValue }>{ stringValue }</ExternalLink>;
```
Comment (simison) [inline projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx:168]: I saw in backend we tried ensurring this has `http(s)` — do we need the same here?

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+const FieldPreview = ( { field, onFilePreview }: FieldPreviewProps ) => {
+	const { label, value } = field;
+	const fieldType = getEffectiveFieldType( field );
+	const icon = getFieldIcon( fieldType );
+
+	const renderFieldValue = () => {
+		// Image select fields
+		if ( fieldType === 'image-select' ) {
+			const choices = ( value as { choices?: unknown[] } )?.choices;
```
[the remark this answers is not in the document]
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx:109]: @copilot Isn't that handled by the rendering component `FieldImageSelect`? Can you double check?

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx (hunk 2 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx:168]: Might be the case, we should only render a link if starts with http(s), let me fix that.

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+		// Handle objects that aren't special types - convert to string representation
+		if ( typeof value === 'object' ) {
+			return JSON.stringify( value );
+		}
+
+		const stringValue = String( value );
+
+		// Emails
+		if ( fieldType === 'email' && EMAIL_REGEX.test( stringValue ) ) {
```
Comment (simison) [inline projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx:158]: This could be `||` to allow old field submissions render email even if we didn't know their type?

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss (hunk 1 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss]: mmm... using the `Icon` component would not be as straightforward as it sounds, we have a diverse way of defining the icons and this is a wee/simple style to get them all aligned in terms of styling?

Comment (simison) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss]: I also just think it's a good habit to use `Icon` in WP UIs for things that contain icons, rather than inlining SVGs.  :-) 

Icon is pretty flexible with what it takes, so my reasoning was that you'd only not simplify CSS, but also the `getIcon()`.

The only thing you likely need to then take into account is the `{src}` structure, probably.

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx (hunk 4 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx:158]: All this render is dependent on the Feedback class already mangling the responses and assigning the field types. Worst case we only print the email address with no decoration. I'd like it to stay restrictive for the same reason: if somehow (or maybe some old entry) gets to this point and even with the checks in place we get something that doesn't look like an email, I'd just print the field value.

Comment (simison) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx:158]: Gotcha 👍 I thought we render both old and new format fields with the same set of if's. :-)

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+		// Handle arrays (e.g., multiple choice selections)
+		if ( Array.isArray( value ) ) {
+			return value.join( ', ' );
+		}
+
+		// Handle objects that aren't special types - convert to string representation
+		if ( typeof value === 'object' ) {
+			return JSON.stringify( value );
+		}
+
```
[the remark this answers is not in the document]
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx:142]: This is part of the next iteration, where we'll deal with special response fields

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss (hunk 1 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss]: Had to wrangle a bit, but now they all use `<Icon />` to render. A preparation function and style for `fill` prop are still needed, but seems more consistent. WDYT?

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss [hunk 6]
```diff
@@ -7,14 +7,7 @@
...
 	color: var(--jp-gray-50, #646970);
-	flex-shrink: 0;
-
-	svg {
-		display: block;
-		width: 24px;
-		height: 24px;
-		fill: currentColor;
-	}
+	fill: var(--jp-gray-50, #646970);
```
Comment (simison) [inline projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss]: Since `Icon` has `fill: currentColor`, I expected the above `colour` to apply it, but if not, then this is fine. :-)

Do you still need the `color` if you have `fill`?

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss (hunk 1 above)
Comment (simison) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss]: Looks great 👍

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss (hunk 6 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/style.scss]: > Since Icon has fill: currentColor, I expected the above colour to apply it, but if not, then this is fine. :-)

It didn't work, my guess is, for it to work, the color should be assigned to some of the parents, but as the icon color is not the same as text color, seemed like too much. It would work if we keep the `color` rule and then just tell `fill` to be `currentColor`.

> Do you still need the color if you have fill?

I "guess" not, there is no text on that element, it was just a _fail safe_. I'll remove it as there is no replacement for the icon.

File: projects/packages/forms/src/dashboard/components/inspector/style.scss [hunk 7]
```diff
@@ -108,22 +108,29 @@
...
 .jp-forms__inbox-response-data {
 	border-top: 1px solid var(--jp-forms-border-color);
-
-	@include mixins.flex-column;
 	font-size: var(--jp-forms-font-size-regular);
 	line-height: 24px;
-	padding: 24px 16px;
-	row-gap: 24px;
-}
+	padding: 0 16px;
```
[the remark this answers is not in the document]
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/inspector/style.scss:114]: Not too aware of how these new "logical" rules are applied, will stick to well-known rules for now.

File: projects/packages/forms/src/dashboard/class-dashboard.php [hunk 8]
```diff
@@ -125,12 +125,13 @@ public function load_admin_scripts() {
...
 		// Pre-fetch the first inbox page so the UI renders instantly on first load.
 		$preload_params = array(
-			'context'  => 'edit',
-			'order'    => 'desc',
-			'orderby'  => 'date',
-			'page'     => 1,
-			'per_page' => 20,
-			'status'   => 'draft,publish',
+			'context'       => 'edit',
+			'fields_format' => 'collection',
```
Comment (enejb) [inline projects/packages/forms/src/dashboard/class-dashboard.php:129]: ❤️

File: projects/packages/forms/src/types/index.ts [hunk 9]
```diff
@@ -317,3 +345,35 @@ export interface FormsConfigData {
 	/** The admin-ajax.php URL for the site. */
 	ajaxUrl?: string;
 }
+
+export type FieldType =
+	| 'name'
```
Comment (enejb) [inline projects/packages/forms/src/types/index.ts:350]: Do we need to keep this update if we introduce a new field?

File: projects/packages/forms/tests/js/dashboard/components/field-preview/field-preview-utils.test.js [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+			];
+
+			invalidEmails.forEach( email => {
+				expect( EMAIL_REGEX.test( email ) ).toBe( false );
+			} );
+		} );
+	} );
+
+	describe( 'FIELD_TYPE_LABEL_PREFIXES', () => {
+		it( 'is an array of tuples with string prefix and field type', () => {
```
Comment (enejb) [inline projects/packages/forms/tests/js/dashboard/components/field-preview/field-preview-utils.test.js:47]: I don't understand this test? What are we testing here?

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/field-preview-utils.ts [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+
+export const EMAIL_REGEX = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
+
+/**
+ * Maps default field label prefixes to their corresponding FieldType.
+ * Used to infer the field type for legacy responses that don't have a type defined.
+ * Labels are stored in lowercase for case-insensitive matching.
+ * Order matters: more specific labels should come before generic ones.
+ */
+export const FIELD_TYPE_LABEL_PREFIXES: Array< [ string, FieldType ] > = [
```
Comment (enejb) [inline projects/packages/forms/src/dashboard/components/response-view/field-preview/field-preview-utils.ts:20]: I think this could be done on the backend. So that it could be used across other types of UI. Also it seems really English dependent.

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+	ExternalLink,
+	Icon,
+	__experimentalHStack as HStack, // eslint-disable-line @wordpress/no-unsafe-wp-apis
+	__experimentalVStack as VStack, // eslint-disable-line @wordpress/no-unsafe-wp-apis
+} from '@wordpress/components';
+/**
+ * Internal dependencies
+ */
+// Field block definitions - used to get consistent icons
+import CheckboxFieldBlock from '../../../../blocks/field-checkbox/index.js';
```
Comment (enejb) [inline projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx]: Instead of doing this. Could we just make sure that the icons are exportable? 
I think this could really increase the size of the dashboard bundle.

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+type FieldPreviewProps = {
+	field: ResponseField;
+	onFilePreview: ( file: FileItem ) => () => void;
+};
+
+const FieldPreview = ( { field, onFilePreview }: FieldPreviewProps ) => {
+	const { label, value, type } = field;
+	// For legacy responses without a proper type (undefined or "basic"), try to infer from label
+	const fieldType: FieldType =
+		type && type !== 'basic' ? type : inferFieldTypeFromLabel( label ) ?? 'text';
```
Comment (enejb) [inline projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx:102]: I would not bother with this... for V1.

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+				<VStack spacing="2" alignment="topLeft">
+					{ ( value as string[] ).map( ( item, index ) => (
+						<Badge key={ index }>{ item }</Badge>
+					) ) }
+				</VStack>
+			);
+		}
+
+		// Handle null/undefined
+		if ( value === null || value === undefined || value === '' ) {
```
Comment (enejb) [inline projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx]: Could we convert the value into string and just check if it is empty ?

File: projects/packages/forms/src/types/index.ts [hunk 15]
```diff
@@ -80,6 +80,34 @@ export type Pattern = {
...
+ */
+export interface ResponseField {
+	/** The field label displayed to users. */
+	label: string;
+	/** The field value. */
+	value: unknown;
+	/** The field type (e.g., 'name', 'email', 'text', 'file', etc.). 'basic' is a legacy value for older responses. */
+	type?: FieldType | 'basic';
+	/** The form field ID from the form schema. */
+	id?: string;
```
Comment (enejb) [inline projects/packages/forms/src/types/index.ts:94]: Do we care about all these (id, key, meta)? How are they important?

File: projects/packages/forms/src/types/index.ts (hunk 9 above)
Comment (CGastrell) [reply projects/packages/forms/src/types/index.ts:350]: Likely, yes.

File: projects/packages/forms/tests/js/dashboard/components/field-preview/field-preview-utils.test.js (hunk 10 above)
Comment (CGastrell) [reply projects/packages/forms/tests/js/dashboard/components/field-preview/field-preview-utils.test.js:47]: Makes sure FIELD_TYPE_LABEL_PREFIXES is an array of tuples (`[ string, string ]`) which is then used to map the inferred field types. The test just makes sure it is well formed.

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/field-preview-utils.ts (hunk 11 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/field-preview-utils.ts:20]: It could. It is English dependent. This is merely a measure to get the right icon for "guessing" the field type. I don't hold high hopes for this as it's basically just mapping default labels to field types. It _works_ but in all honesty it shouldn't even be there. Worst case, the field type defaults to Text Field icon.
Passing it down from backend and making it translatable notches the complexity a bit, if we are going to do it, I'd leave it to a follow up.

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx (hunk 13 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx:102]: Then all the other measures would be obsolete as well (prefix tuples, test). Worst case, is a good way of defaulting to some icon, then fallback to Text Field icon. But if you're paramount about it, then we can get rid of this and the PREFIX array, they are all for there for the same reason.

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx (hunk 14 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx]: It is done at line 144, just making sure it has some content when needed. I think it can be removed, most of this is taken care on backend anyways, this is simply fail safe mechanics. Here's the change for a simpler approach https://github.com/Automattic/jetpack/pull/46706/commits/8b3e986c2b39b9b439010cf256c77d00b8b83c4b

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx (hunk 12 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx]: I thought these would be tree-shaken. I guess we could export a simple object `{ src: icon-here }` on all those blocks and then just import those directly. Good catch, let me see to it.

File: projects/packages/forms/src/types/index.ts (hunk 15 above)
Comment (CGastrell) [reply projects/packages/forms/src/types/index.ts:94]: At the moment we only use `key`, in the past having the `id`s has proven valuable but it has no use right now. Same goes for `meta`. I can remove those from the typing, but feels natural to have the entire object typed on frontend as it is on backend. The props are optional so the format could be capped if we need to (and it would just be a change on backend, no client update needed).

File: projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx (hunk 12 above)
Comment (CGastrell) [reply projects/packages/forms/src/dashboard/components/response-view/field-preview/index.tsx]: https://github.com/Automattic/jetpack/pull/46706/commits/7df6497a96ba3db71b12bef8d1b594a5dd642bf2 and https://github.com/Automattic/jetpack/pull/46706/commits/700923f4a137032cad79c7e714455ffe38793f20
